components/object_store/src/obkv/mod.rs

```diff
@@ -0,0 +1,929 @@
+    }
+}
+
+#[async_trait]
+impl<T: TableKv> ObjectStore for ObkvObjectStore<T> {
+    async fn put(&self, location: &Path, bytes: Bytes) -> Result<()> {
+        let instant = Instant::now();
+
+        self.check_size(&bytes)
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        // Use `put_multipart` to implement `put`.
+        let (_upload_id, mut multipart) = self.put_multipart(location).await?;
+        multipart
+            .write(&bytes)
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+        // Complete stage: flush buffer data to obkv, and save meta data
+        multipart
+            .shutdown()
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+        debug!(
+            "ObkvObjectStore put operation, location:{},cost:{}",
+            location,
+            instant.elapsed().as_millis()
+        );
+        Ok(())
+    }
+
+    async fn put_multipart(
+        &self,
+        location: &Path,
+    ) -> Result<(MultipartId, Box<dyn AsyncWrite + Unpin + Send>)> {
+        let instant = Instant::now();
+
+        let upload_id = self.current_upload_id.fetch_add(1, Ordering::Relaxed);
+        let multi_part_id = format!("{upload_id}");
+        let table_name = self.pick_shard_table(location);
+
+        let upload = ObkvMultiPartUpload {
+            location: location.clone(),
+            upload_id: multi_part_id.clone(),
+            table_name: table_name.to_string(),
+            size: AtomicU64::new(0),
+            client: Arc::clone(&self.client),
+            part_size: self.part_size,
+            meta_manager: self.meta_manager.clone(),
+        };
+        let multi_part_upload =
+            CloudMultiPartUpload::new(upload, self.max_upload_concurrency, self.part_size);
+
+        debug!(
+            "ObkvObjectStore put_multipart operation, location:{}, table_name:{}, cost:{}",
+            location,
+            table_name,
+            instant.elapsed().as_millis()
+        );
+        Ok((multi_part_id, Box::new(multi_part_upload)))
+    }
+
+    async fn abort_multipart(&self, location: &Path, multipart_id: &MultipartId) -> Result<()> {
+        let instant = Instant::now();
+
+        let table_name = self.pick_shard_table(location);
+
+        // Before aborting multipart, we need to delete all data parts and meta info.
+        // Here to delete data with path `location` and multipart_id
+        let scan_context: ScanContext = ScanContext {
+            timeout: time::Duration::from_secs(meta::SCAN_TIMEOUT_SECS),
+            batch_size: meta::SCAN_BATCH_SIZE,
+        };
+
+        let prefix: String = format!("{}@{}@", location.as_ref(), multipart_id);
```

Comment:
Maybe the format of the prefix should be handled by `ObkvMultiPartUpload`, and there is no need to delcare the type for it:
```suggestion
        let prefix = format!("{location}@{multipart_id}@");
```